### PR TITLE
Clear finished function

### PR DESCRIPTION
**Pull request by nightism**

### CODE REVIEW

**Comment by nightism** [comment]
@pyokagan please review it first, see whether this is critical issue

**Comment by pyokagan** [comment]
@nightism Can try rebasing onto master first?

File: src/main/java/seedu/address/model/TaskBook.java [hunk 1]
```diff
@@ -67,6 +68,14 @@ public FloatingTask removeFloatingTask(int index) {
         return floatingTasks.remove(index);
     }
 
+    /**
+     * Remove all the floating tasks that satisfy the given predicate.
+     * @param predicate The filter used to judge whether a floating task needs to be deleted or not.
+     */
+    public boolean removeFloatingTasks(Predicate<? super FloatingTask> predicate) {
+        return floatingTasks.removeIf(predicate);
+    }
```
Comment (pyokagan) [inline src/main/java/seedu/address/model/TaskBook.java:77]: These TaskBook functions do not even seem to be used at all?

File: src/main/java/seedu/address/model/WorkingTaskBook.java [hunk 2]
```diff
@@ -90,6 +90,27 @@ public FloatingTask removeFloatingTask(int workingIndex) throws IllegalValueExce
...
     }
 
+    /**
+     * Remove all the floating tasks that satisfy the given predicate.
+     * @param predicate The filter used to judge whether a floating task needs to be deleted or not.
+     * @throws IllegalValueException Will be throw during iteration if internal error happens.
+     */
+    public void removeFloatingTasks(Predicate<? super FloatingTask> predicate) {
+        int iterator = 1;
+        while (iterator < workingFloatingTasks.size()) {
```
Comment (pyokagan) [inline src/main/java/seedu/address/model/WorkingTaskBook.java:100]: Doesn't this mean we only are able to delete tasks in the current working list (which is a subset of the whole task book)? I thought you wanted clear to affect the whole task book?

**Comment by pyokagan** [comment]
@nightism So I wonder if it is easier to just do the deletions on the task book side, and then repopulate the working task lists from it, while using the same filter.

File: src/main/java/seedu/address/model/WorkingTaskBook.java (hunk 2 above)
Comment (nightism) [reply src/main/java/seedu/address/model/WorkingTaskBook.java:100]: i just followed your advice from my last pr...

Comment (nightism) [reply src/main/java/seedu/address/model/WorkingTaskBook.java:100]: for now it just clear what the user can see

**Comment by nightism** [comment]
new PR created @pyokagan

**Pull request closed without merging**